internal/plan/plan_test.go

```diff
@@ -1017,3 +1043,91 @@ func (s *S) TestMarshalLayer(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(string(out), Equals, string(layerBytes))
 }
+
+func (s *S) TestLogTargetSelection(c *C) {
+	layerYAML := `
+log-targets:
+  optout:
+    type: loki
+    location: foo
+    override: merge
+  optin:
+    type: loki
+    location: foo
+    override: merge
+    selection: opt-in
+  disabled:
+    type: loki
+    location: foo
+    override: merge
+    selection: disabled
+
+services:
+  svc1:
+    command: cmd
+    override: merge
+  svc2:
+    command: cmd
+    override: merge
+    log-targets:
+  svc3:
+    command: cmd
+    override: merge
+    log-targets:
+      - optout
+  svc4:
+    command: cmd
+    override: merge
+    log-targets:
+      - optin
+  svc5:
+    command: cmd
+    override: merge
+    log-targets:
+      - disabled
+  svc6:
+    command: cmd
+    override: merge
+    log-targets:
+      - optout
+      - optin
+  svc7:
+    command: cmd
+    override: merge
+    log-targets:
+      - optout
+      - disabled
+  svc8:
+    command: cmd
+    override: merge
+    log-targets:
+      - optin
+      - disabled
+  svc9:
+    command: cmd
+    override: merge
+    log-targets:
+      - optout
+      - optin
+      - disabled
+`
+	expectedTargets := map[string][]string{
+		"svc1": {"optout"},
+		"svc2": {"optout"}, // or should it be none?
+		"svc3": {"optout"},
+		"svc4": {"optin"},
+		"svc5": {},
+		"svc6": {"optout", "optin"},
+		"svc7": {"optout"},
+		"svc8": {"optin"},
+		"svc9": {"optout", "optin"},
+	}
+
+	layer, err := plan.ParseLayer(0, "", []byte(layerYAML))
+	c.Assert(err, IsNil)
+	layer, err = plan.CombineLayers(layer)
+	c.Assert(err, IsNil)
+	for serviceName, service := range layer.Services {
+		c.Check(service.LogTargets, testutil.SameContents, expectedTargets[serviceName])
```

Comment:
Instead of writing an entirely new checker, why don't we just make the code that finalizes the log targets call `sort.Strings()` on it? (Or we could sort it here, but I think stable ordering is good in any case.)